Library/Homebrew/dev-cmd/bump.rb

```diff
@@ -373,13 +374,17 @@ def retrieve_versions_by_arch(formula_or_cask:, repositories:, name:)
           new_version.general.to_s
         end
 
-        duplicate_pull_requests = unless args.no_pull_requests?
-          retrieve_pull_requests(formula_or_cask, name, version: pull_request_version)
-        end.presence
+        if !args.no_pull_requests? && (new_version != current_version)
+          duplicate_pull_requests = retrieve_pull_requests(
+            formula_or_cask,
+            name,
+            version: pull_request_version,
+          )
 
-        maybe_duplicate_pull_requests = if !args.no_pull_requests? && duplicate_pull_requests.blank?
-          retrieve_pull_requests(formula_or_cask, name)
-        end.presence
+          maybe_duplicate_pull_requests = if duplicate_pull_requests.blank?
```

Comment:
```suggestion
          maybe_duplicate_pull_requests = if duplicate_pull_requests.nil?
```

Nit: `#blank?` should no longer be necessary here as `retrieve_pull_requests` returns nil when there are no pull requests.